### PR TITLE
Add prettier

### LINKED ISSUE #3: Add prettier
Setup prettier on the whole codebase.

### PR DESCRIPTION
**Pull request by agnesesterhuizen**
Hey Ian! I've added prettier with some basic rules and a npm script to run it on the source and example code. I have also added it to run with the npm prepare script but I wasn't completely sure if this is the right stage to run it. Please let me know if you think there's anything I should change or add or if there are any other issues. 

Fixes #3

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "useTabs": false,
+  "printWidth": 80,
+  "tabWidth": 2,
+  "singleQuote": true,
+  "trailingComma": "all"
+}
```
Comment (ianobermiller) [inline .prettierrc:9]:
```
{
  "bracketSpacing": false,
  "jsxBracketSameLine": true,
  "printWidth": 80,
  "singleQuote": true,
  "tabWidth": 2,
  "trailingComma": "all",
  "useTabs": false
}
```

Comment (ianobermiller) [reply .prettierrc:9]: Let's also add the rest of the config options. In particular I don't want `bracketSpacing` :)

File: package.json [hunk 2]
```diff
@@ -15,9 +15,10 @@
   "scripts": {
     "test": "cross-env CI=1 react-scripts-ts test --env=jsdom",
     "test:watch": "react-scripts-ts test --env=jsdom",
+    "prettier": "prettier --config .prettierrc --write \"{src,example}/**/*.{js,ts,tsx}\"",
     "build": "rollup -c",
     "start": "rollup -c -w",
-    "prepare": "yarn run build",
+    "prepare": "yarn run prettier && yarn run build",
```
Comment (ianobermiller) [inline package.json:21]: lgtm

**Review by ianobermiller — CHANGES_REQUESTED** [review]
Thanks for picking up this task! Sorry for the churn! Just fix the config and rebase and you should be good to go!

**Comment by agnesesterhuizen** [comment]
@ianobermiller Sorry for the delay in working on this, I've updated the config now and fixed the merge conflicts. Please let me know if theres anything else you think needs to change.

**Comment by ianobermiller** [comment]
No worries, it's open source, no rush! Thanks for taking this on and fixing it up!

**Pull request merged**